driver/csiplugin/controllerserver.go

```diff
@@ -1747,20 +1838,52 @@ func (cs *ScaleControllerServer) CreateSnapshot(ctx context.Context, req *csi.Cr
 		}
 	}
 
-	if filesetResp.Config.ParentId > 0 {
-		return nil, status.Error(codes.InvalidArgument, fmt.Sprintf("CreateSnapshot - volume [%s] - Volume snapshot can only be created when source volume is independent fileset", volID))
+	if volumeIDMembers.StorageClassType != STORAGECLASS_ADVANCED {
+		if filesetResp.Config.ParentId > 0 {
+			return nil, status.Error(codes.InvalidArgument, fmt.Sprintf("CreateSnapshot - volume [%s] - Volume snapshot can only be created when source volume is independent fileset", volID))
+		}
 	}
+
 	filesetName := filesetResp.FilesetName
-	sLinkRelPath := strings.Replace(volumeIDMembers.Path, cs.Driver.primary.PrimaryFSMount, "", 1)
-	sLinkRelPath = strings.Trim(sLinkRelPath, "!/")
+	relPath := ""
+	if volumeIDMembers.StorageClassType == STORAGECLASS_ADVANCED {
+		glog.V(3).Infof("CreateSnapshot - creating snapshot for advanced storageClass")
+		primaryConn, isprimaryConnPresent := cs.Driver.connmap["primary"]
+		if !isprimaryConnPresent {
+			glog.Errorf("CreateSnapshot - unable to get connector for primary cluster")
+			return nil, status.Error(codes.Internal, "CreateSnapshot - unable to find primary cluster details in custom resource")
+		}
+		mountInfo, err := primaryConn.GetFilesystemMountDetails(filesystemName)
+		if err != nil {
+			return nil, status.Error(codes.Internal, fmt.Sprintf("CreateSnapshot - unable to get mount info for FS [%v] in primary cluster", filesystemName))
+		}
+		relPath = strings.Replace(volumeIDMembers.Path, mountInfo.MountPoint, "", 1)
+	} else {
+		glog.V(3).Infof("CreateSnapshot - creating snapshot for classic storageClass")
+		relPath = strings.Replace(volumeIDMembers.Path, cs.Driver.primary.PrimaryFSMount, "", 1)
+	}
+	relPath = strings.Trim(relPath, "!/")
 
 	/* Confirm it is same fileset which was created for this PV */
-	pvName := filepath.Base(sLinkRelPath)
+	pvName := filepath.Base(relPath)
 	if pvName != filesetName {
 		return nil, status.Error(codes.Internal, fmt.Sprintf("CreateSnapshot - PV name from path [%v] does not match with filesetName [%v].", pvName, filesetName))
 	}
 
+	if volumeIDMembers.StorageClassType == STORAGECLASS_ADVANCED {
+		filesetName = volumeIDMembers.ConsistencyGroup
+	}
+
 	snapName := req.GetName()
+	snapWindow := ""
+	if volumeIDMembers.StorageClassType == STORAGECLASS_ADVANCED {
+		snapParams := req.GetParameters()
+		snapWindowSpecified := false
+		snapWindow, snapWindowSpecified = snapParams[connectors.UserSpecifiedSnapWindow]
+		if !snapWindowSpecified {
+			return nil, status.Error(codes.Internal, fmt.Sprintf("snapWindow not specified for fileset[%s:%s]", filesetResp.FilesetName, filesystemName))
```

Comment:
If snapWindow is not specified then rather than returning error use snapWindow=30

Reply:
Done